docs/content/sips/xxx-clock-trigger.md

```diff
@@ -0,0 +1,101 @@
+title = "SIP xxx - Clock Trigger"
+template = "main"
+date = "2022-05-23T16:46:52Z"
+---
+
+Summary: Support a clock trigger for Spin applications.
+
+Owner: brian.hardock@fermyon.com
+
+Created: May 23, 2022
+
+Updated: May 23, 2022
+
+## Background
+
+This proposal aims to introduce a new `Clock` trigger type whereby components can define their own schedule to be automatically executed at specific times.
+As an example, an application developer might use this trigger type to implement a component that sends out a daily report email or to update cached data every 30 seconds.
+
+## Proposal
+
+Components using the clock trigger type must implement the following WebAssembly interface:
+
+```fsharp
+// wit/ephemeral/spin-clock.wit
+
+// Context for executing a clock triggered component
+record clock-event {
+    // Time at which this event fired represented as a unix timestamp
+    timestamp: u64
+    // Indicates if executing on start
+    on-start: bool
+}
+
+// The entrypoint for a clock trigger component 
+handle-clock-event: function(event: clock-event) -> expected<_, string>
```

Comment:
Is the idea to have more specific errors here in the future?

Reply:
Yes

Reply:
What should we standardize in for wit errors? `varient error`? Maybe with an optional `other(string)` if needed?

Reply:
Particularly the `other(string)` variant for the catch case when errors are so enumerable.